blocks/components/editable/index.js

```diff
@@ -44,9 +44,9 @@ export default class Editable extends wp.element.Component {
 	onSetup( editor ) {
 		this.editor = editor;
 		editor.on( 'init', this.onInit );
-		editor.on( 'focusout', this.onChange );
 		editor.on( 'NewBlock', this.onNewBlock );
 		editor.on( 'focusin', this.onFocus );
+		editor.on( 'input', this.onChange );
```

Comment:
Tested this branch and I'm already having focus jumps while typing. These are really hard to solve in my exp with the per-block prototype.